Lesson 3

slade1987 opened the pull request.

slade1987 committed:
Lesson 3
Algorim
Paranych Ivan Vladimirovich

DmitryChitalov commented on Урок 3. Практическое задание/task_1.py:
```diff
@@ -28,3 +28,94 @@
+mas = []
+
+
+def countTime(funct):
```
countTime
нарушение пеп-8 в именах

DmitryChitalov commented on Урок 3. Практическое задание/task_1.py:
```diff
@@ -28,3 +28,94 @@
+реализация хэш-таблицы должна учитывать возможность появления коллизий – ситуаций, 
+когда разные ключи имеют одинаковое значение хэша. Должен быть способ вставки и 
+извлечения элементов с учётом коллизий, что накладывает дополнительное временя на
+'внутренню работу' c данными
```
выполнено

DmitryChitalov commented on Урок 3. Практическое задание/task_1.py:
```diff
@@ -28,3 +28,94 @@
+
+"""
+Получения элементов из словаря быстрее так как это хеш таблица
+"""
```
выполнено

DmitryChitalov commented on Урок 3. Практическое задание/task_2.py:
```diff
@@ -22,3 +22,51 @@
+            self.con.execute("""CREATE TABLE hash(login text,pass text, finalHash text)""")
+
+    def getName(self, name, password):
+        hash = hashlib.sha256(str(name).encode('utf-8') + str(password).encode('utf-8'))
```
hash = hashlib.sha256(str(name).encode('utf-8') + str(password).encode('utf-8'))
password = input("Введите пароль")

дублирование
нужно вынести в отд. ф-цию
посмотрите пример

DmitryChitalov commented on Урок 3. Практическое задание/task_3.py:
```diff
@@ -22,3 +22,17 @@
+    for j in range(len(a) + 1):
+        k.add((a[i:j + i + 1],hashlib.sha256(str(a[i:j + i + 1]).encode('utf-8')).hexdigest()))
+
+print(k)
```
выполнено

DmitryChitalov commented on Урок 3. Практическое задание/task_4.py:
```diff
@@ -15,3 +15,22 @@
+getURL('ya.ru')
+getURL('yandex.ru')
+getURL('yandex.ru')
+print(webHash)
```
выполнено

slade1987 committed:
Сделал реформат кода согласно pep-8

slade1987 committed:
Сделал реформат кода согласно pep-8
Вынес часть кода в отдельную функцию
hash = hashlib.sha256(str(name).encode('utf-8') + str(password).encode('utf-8'))
